src/v2/Components/Authentication/Validators.tsx

```diff
@@ -35,13 +35,18 @@ export const LoginValidator = Yup.object().shape({
 })
 
 export const MobileSignUpValidator = {
-  email: Yup.object().shape({ email, accepted_terms_of_service }),
+  email: Yup.object().shape({ accepted_terms_of_service, email }),
   name: Yup.object().shape({ name }),
   password: Yup.object().shape({ password }),
 }
 
 export const MobileLoginValidator = {
   email: Yup.object().shape({ email }),
-  password: Yup.object().shape({ password: loginPassword }),
   otpAttempt: Yup.object().shape({ otp_attempt: otpAttempt }),
+  password: Yup.object().shape({ password: loginPassword }),
 }
+
+export const ChangeEmailValidator = Yup.object().shape({
```

Comment:
Yup is required by formik, and can validate a field on the presence of another field, but the lib's error handling causes issues when we want to conditionally show/hide the password field. Using a manual password error rather than a formik error allows us to avoid resetting the form states manually if the email address is changed and then reverted before saving. I've updated this naming to be more clear.

Reply:
This is the validation i was messing with via yup:
```
  password: Yup.string().when("email", {
    is: value => value && value.length > 0,
    otherwise: Yup.string(),
    then: Yup.string().required(
      "Password is required when changing email address."
    )
  }),
```

Reply:
Oh cool! Yea, that's sort of how I was imagining it!